py3status/modules/wwan_status_nm.py

```diff
@@ -0,0 +1,160 @@
+        (default False)
+    format_down: What to display when the modem is not plugged in.
+        (default 'WWAN: {state} - {operator} {netgen} ({signal}%)')
+    format_error: What to display when modem can't be accessed.
+        (default 'WWAN: {error}')
+    format_up: What to display upon regular connection
+        (default 'WWAN: {state} - {operator} {netgen} ({signal}%)')
+    modem: The modem device to use. If None
+        will use first find modem or
+        use 'busctl introspect org.freedesktop.ModemManager1 \
+            /org/freedesktop/ModemManager1/Modem/0'
+        and read '.EquipmentIdentifier')
+        (default None)
+
+Color options:
+    color_bad: Error or no connection
+    color_degraded: Low generation connection eg 2G
+    color_good: Good connection
+
+Requires:
+    ModemManager
+    NetworkManager
+    pydbus
+
+@author Cyril Levis <levis.cyril@gmail.com>
+
+SAMPLE OUTPUT
+{'color': '#00FF00', 'full_text': u'WWAN: Connected - Bouygues Telecom 4G (19%)'}
+
+off
+{'color': '#FF0000', 'full_text': u'WWAN: Disconnected - Bouygues Telecom 4G (12%)'}
+"""
+
+from pydbus import SystemBus
+
+STRING_WRONG_MODEM = "wrong or any modem"
+STRING_UNKNOW_OPERATOR = "unknow operator"
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    cache_timeout = 5
+    consider_3G_degraded = False
+    format_down = 'WWAN: {state} - {operator} {netgen} ({signal}%)'
+    format_error = 'WWAN: {error}'
+    format_up = 'WWAN: {state} - {operator} {netgen} ({signal}%)'
+    modem = None
+
+    def wwan_status_nm(self):
+        response = {}
+        response['cached_until'] = self.py3.time_in(self.cache_timeout)
+
+        states = {10: "Connecting", 11: "Connected"}
+
+        # https://www.freedesktop.org/software/ModemManager/api/1.0.0/ModemManager-Flags-and-Enumerations.html#MMModemAccessTechnology
+        speed = {
+            16384: 'LTE',
+            8192: 'EVDOB',
+            4096: 'EVDOA',
+            2048: 'EVDO0',
+            1024: '1XRTT',
+            512: 'HSPA+',
+            256: 'HSPA',
+            128: 'HSUPA',
+            64: 'HSDPA',
+            32: 'UMTS',
+            16: 'EDGE',
+            8: 'GPRS',
+            4: 'GSM_COMPACT',
+            2: 'GSM',
+            0: 'POTS'
+        }
+
+        bus = SystemBus()
+        for id in range(0, 20):
+
+            # find the modem
+            device = 'Modem/' + str(id)
+
+            try:
+                proxy = bus.get('.ModemManager1', device)
+
+                # we can maybe choose another selector
+                eqid = str(proxy.EquipmentIdentifier)
+
+                if self.modem is None or self.modem == eqid:
+
+                    # get status informations
+                    status = proxy.GetStatus()
+
+                    # start to build return data dict
+                    data = {
+                        'state': 'Disconnected',
+                        'signal': status['signal-quality'][0],
```

Comment:
What is in `status`?

Reply:
example:
{'state': 11, 'signal-quality': (29, False), 'current-bands': [0], 'access-technologies': 16384, 'm3gpp-registration-state': 1, 'm3gpp-operator-code': '20820', 'm3gpp-operator-name': 'Bouygues Telecom'